pixi/src/ui/PageExperience.js

```diff
@@ -0,0 +1,69 @@
+// Copyright 2020 The AMPHTML Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import PageExperienceCheck from '../checks/PageExperienceCheck.js';
+import CoreWebVitalsReport from './report/CoreWebVitalsReport.js';
+
+export default class PageExperience {
+  constructor() {
+    this.input = document.getElementById('input-field');
+    this.submit = document.getElementById('input-submit');
+    this.submit.addEventListener('click', this.onSubmitUrl.bind(this));
+
+    this.reportViews = {};
+  }
+
+  isValidURL(inputUrl) {
```

Comment:
Btw one option to have this validate on its own would be to use add the `required` attribute to the input field and use it with a `<form>`